src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/ILspServices.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Immutable;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+// TODO: Does this need a GetRequiredServices<T> or is that too specific to some implementations?
```

Comment:
Might even be able to reduce the scope even more if we want to follow in [IServiceProvider](https://github.com/dotnet/runtime/blob/67f0ee59a5118418a068fa2fbb77834b62038ff6/src/libraries/System.ComponentModel/src/System/IServiceProvider.cs)'s footsteps

Reply:
@NTaylorMullen taking the implementation down to just `GetService<T>` would mean that we're not able to do Handler discovery off of ILspServices because we need some way to either get ALL the services registered, or to get multiple services which are registered for the same type.